sphinxgallery/gen_rst.py

```diff
@@ -118,133 +78,161 @@ def flush(self):
 HLIST_IMAGE_TEMPLATE = """
     *
 
-      .. image:: images/%s
+      .. image:: /%s
             :scale: 47
 """
 
 SINGLE_IMAGE = """
-.. image:: images/%s
+.. image:: /%s
     :align: center
 """
 
+CODE_OUTPUT = """**Script output**:\n
+.. rst-class:: sphx-glr-script-out
+
+  ::
+
+    {0}\n"""
+
 
-def extract_docstring(filename, ignore_heading=False):
-    """ Extract a module-level docstring, if any
+def split_code_and_text_blocks(source_file):
+    """Return list with source file separated into code and text blocks.
+
+    Returns
+    -------
+    blocks : list of (label, content)
+        List where each element is a tuple with the label ('text' or 'code'),
+        and content string of block.
     """
-    lines = open(filename).readlines()
-    start_row = 0
-    if lines[0].startswith('#!'):
-        lines.pop(0)
-        start_row = 1
-    docstring = ''
-    first_par = ''
-    line_iterator = iter(lines)
-    tokens = tokenize.generate_tokens(lambda: next(line_iterator))
-    for tok_type, tok_content, _, (erow, _), _ in tokens:
-        tok_type = token.tok_name[tok_type]
-        if tok_type in ('NEWLINE', 'COMMENT', 'NL', 'INDENT', 'DEDENT'):
+    f = open(source_file)
+
+    blocks = []
+    block = ''
+
+    has_header_string = 0
+    continue_text = False
+    for line in f:
+        # python docstring
+        if line.startswith('"""') and not has_header_string:
```

Comment:
There are a few things to this. Indeed only a single docstring header will be supported and that uses the triple quotes.
Now this function is indeed quite fragile as is was designed yesterday and for our specific syntax case. Although the notebook example does provide some tricky syntax from time to time.
There was in between a new block splitter based on the tokenize, but that splitted based on strings